src/ResourceTemplate/RelativeResourceTemplate.php

```diff
@@ -149,24 +155,73 @@ public function matches($path)
     public function match($path)
     {
         // High level strategy for matching:
-        // - Break $path into pieces based on '/'
         // - Build a list of Segments from our template, where any variable segments are
         //   flattened into a single, non-nested list
+        // - Break $path into pieces based on '/'.
+        //     - Use the segments to further subdivide the pieces using any applicable non-slash separators.
         // - Match pieces of the path with Segments in the flattened list
 
-        $pathPieces = explode('/', $path);
-        $pathPiecesCount = count($pathPieces);
-
         // In order to build correct bindings after we match the $path against our template, we
         // need to (a) calculate the correct positional keys for our wildcards, and (b) maintain
         // information about the variable identifier of any flattened segments. To do this, we
         // build a list of [string, Segment] tuples, where the string component is the appropriate
         // key.
         $keySegmentTuples = self::buildKeySegmentTuples($this->segments);
+
         $flattenedKeySegmentTuples = self::flattenKeySegmentTuples($keySegmentTuples);
         $flattenedKeySegmentTuplesCount = count($flattenedKeySegmentTuples);
         assert($flattenedKeySegmentTuplesCount > 0);
 
+
```

Comment:
Would we please be able to drop this whitespace?

Reply:
Done.